### ISSUE TITLE
Training COCO from scratch box loss 0.00

### ISSUE THREAD

**Issue by BradleyBrown19** [google/automl#713]
I started training coco training from scratch and have been getting only det_loss, the box_loss is always 0.00.

I used the following to make the shards:

`!PYTHONPATH=".:$PYTHONPATH"  python dataset/create_coco_tfrecord.py \
    --image_dir=train2017 \
    --caption_annotations_file=annotations/captions_train2017.json \
    --output_file_prefix=tfrecord/train \
    --num_shards=32`

and then trained using the following command:

`!mkdir /tmp/model_dir/'

'
!python main.py --mode=train \
    --training_file_pattern=tfrecord/{file_pattern} \
    --validation_file_pattern=tfrecord/{file_pattern_val} \
    --model_name={MODEL} \
    --model_dir=/tmp/model_dir/baseline_{MODEL}-finetune2 \
    --train_batch_size=8 \
    --eval_batch_size=8 --eval_samples={val_images_per_epoch}  \
    --num_examples_per_epoch={images_per_epoch}  --num_epochs=90 \
    --eval_after_training True `

where 

`file_pattern = 'train-*-of-00032.tfrecord'  # @param'

file_pattern_val = 'val-*-of-00032.tfrecord'  # @param'
images_per_epoch = 118287'
val_images_per_epoch = 5000'
MODEL = 'efficientdet-d1'`

This is my config:

`{'name': 'efficientdet-d1', 'act_type': 'swish', 'image_size': (640, 640), 'target_size': None, 'input_rand_hflip': True, 'jitter_min': 0.1, 'jitter_max': 2.0, 'autoaugment_policy': 'v0', 'use_augmix': False, 'augmix_params': (3, -1, 1), 'sample_image': None, 'num_classes': 90, 'seg_num_classes': 3, 'heads': ['object_detection'], 'skip_crowd_during_training': True, 'label_map': 'coco', 'max_instances_per_image': 100, 'regenerate_source_id': False, 'min_level': 3, 'max_level': 7, 'num_scales': 3, 'aspect_ratios': [(1.0, 1.0), (1.4, 0.7), (0.7, 1.4)], 'anchor_scale': 4.0, 'is_training_bn': True, 'momentum': 0.9, 'optimizer': 'sgd', 'learning_rate': 0.08, 'lr_warmup_init': 0.008, 'lr_warmup_epoch': 1.0, 'first_lr_drop_epoch': 200.0, 'second_lr_drop_epoch': 250.0, 'poly_lr_power': 0.9, 'clip_gradients_norm': 10.0, 'num_epochs': 90, 'data_format': 'channels_last', 'label_smoothing': 0.0, 'alpha': 0.25, 'gamma': 1.5, 'delta': 0.1, 'box_loss_weight': 50.0, 'iou_loss_type': None, 'iou_loss_weight': 1.0, 'weight_decay': 4e-05, 'strategy': None, 'mixed_precision': False, 'box_class_repeats': 3, 'fpn_cell_repeats': 4, 'fpn_num_filters': 88, 'separable_conv': True, 'apply_bn_for_resampling': True, 'conv_after_downsample': False, 'conv_bn_act_pattern': False, 'nms_configs': {'method': 'gaussian', 'iou_thresh': None, 'score_thresh': None, 'sigma': None, 'max_nms_inputs': 0, 'max_output_size': 100}, 'fpn_name': None, 'fpn_weight_method': None, 'fpn_config': None, 'survival_prob': None, 'img_summary_steps': None, 'lr_decay_method': 'cosine', 'moving_average_decay': 0.9998, 'ckpt_var_scope': None, 'skip_mismatch': True, 'backbone_name': 'efficientnet-b1', 'backbone_config': None, 'var_freeze_expr': None, 'use_keras_model': True, 'resnet_depth': 50, 'model_name': 'efficientdet-d1', 'iterations_per_loop': 100, 'model_dir': '/tmp/model_dir/baseline_efficientdet-d1-finetune2', 'num_shards': 8, 'num_examples_per_epoch': 118287, 'backbone_ckpt': '', 'ckpt': None, 'val_json_file': None, 'testdev_dir': None, 'mode': 'train'}`

Please let me know if anyone knows any issue or if there is any more information I can provide. Thanks!

**Comment by InfiniteLife** [comment]
May be some weights are left not initialized? You can try to loop through all model weights to check if there is any zeros weights.

**Comment by BradleyBrown19** [comment]
All the weights are initialized, does anyone have any other ideas?

**Comment by kartik4949** [comment]
@mingxingtan num classes look 90 in the above , which  is wrong it should be 91,right?
also @BradleyBrown19 please refer #676

**Comment by BradleyBrown19** [comment]
Thank you @kartik4949. I think my issues is that I did not pass the object annotations flag to my tfrecords. Adding this line to the records command seems to have fixed the problem:

`--object_annotations_file=annotations/instances_val2017.json \`

Also, thank you for pointing me to that thread, I have set use_keras_model to false and num_classes to 91. Hopefully this does the trick!
